wasm/jit/engine.go

```diff
@@ -420,17 +422,23 @@ func (e *engine) exec(f *compiledWasmFunction) {
 	}
 }
 
-func (e *engine) memoryGrow(m *wasm.MemoryInstance, newPages uint64) {
+func (e *engine) builtinFunctionMemoryGrow(f *compiledWasmFunction) {
+	newPages := e.pop()
 	max := uint64(math.MaxUint32)
-	if m.Max != nil {
-		max = uint64(*m.Max) * wasm.PageSize
+	if f.memory.Max != nil {
+		max = uint64(*f.memory.Max) * wasm.PageSize
 	}
 	// If exceeds the max of memory size, we push -1 according to the spec
-	if uint64(newPages*wasm.PageSize+uint64(len(m.Buffer))) > max {
+	if uint64(newPages*wasm.PageSize+uint64(len(f.memory.Buffer))) > max {
```

Comment:
could newPages be zero? what should we do if that were the case?

Reply:
zero is valid and even in that case this function works as expected. Added the test case: https://github.com/tetratelabs/wazero/pull/85/commits/19f32ca4cdc9d55c4219d0590fbd5ca68ef608c8